c++/src/kj/exception.h

```diff
@@ -133,6 +134,25 @@ class Exception {
   KJ_NOINLINE void addTraceHere();
   // Adds the location that called this method to the stack trace.
 
+  using DetailTypeId = unsigned long long;
+
+  kj::Maybe<kj::ArrayPtr<const byte>> getDetail(DetailTypeId typeId) const;
+  void setDetail(DetailTypeId typeId, kj::Array<byte> value);
+  kj::Maybe<kj::Array<byte>> releaseDetail(DetailTypeId typeId);
+  // Details: Arbitrary extra information can be added to an exception. Applications can define
+  // any kind of detail they want, but it must be serializable to bytes so that it can be logged
+  // and transmitted over RPC.
+  //
+  // Every type of detail must have a unique ID, which is a 64-bit integer. It's suggested that
+  // you use `capnp id` to generate these.
+  //
+  // It is expected that exceptions will rarely have more than one or two details, so the
+  // implementation uses a flat array with O(n) lookup.
+  //
+  // The main use case for details is to be able to tunnel exceptions of a different type through
+  // KJ / Cap'n Proto. In particular, Cloudflare Workers commonly has to convert a JavaScript
+  // exception to KJ and back. The exception is serialized using V8 serialization.
```

Comment:
Right. For the moment, I didn't need this, and I kind of like that applications can only query for the types they actually know about, rather than trying to explore unrecognized types.

That said, I suppose this may have to change when we extend this to transit RPC correctly. At present the details will be lost when crossing RPC -- but for the present use case, that doesn't matter.